lib/index.js

```diff
@@ -27,9 +28,12 @@ if (options.version) {
 }
 
 var updateIndex = function(data, callback) {
-    var index = path.join(options.dir, 'index.json');
-    fs.readFile(index, 'utf8', function(err, d) {
-        if (d) {
+    hooks.globalIndexJson(data, callback, function() {
+        var index = path.join(options.dir, 'index.json');
+        fs.readFile(index, 'utf8', function(err, d) {
+            if (!d) {
+                return callback(d);
+            }
```

Comment:
I think that I didn't handle the error case here is to ensure that it is always written even if it doesn't exist.